File: frontend/shared/types/query.ts
export interface QueryOptions {
  pagination: {
    page: number;
    pageSize: number;
  };
  keyword?: string;
  filters?: {
    [key: string]: string | string[] | number | boolean | null | undefined;
  };
  sorting?: {
    field: string;
    order: 'asc' | 'desc';
  };
}

export interface PaginatedResponse<T> {
  data: T[];
  meta: {
    rowCount: number;
    pageCount: number;
    filters?: {
      [key: string]: string[];
    };
  };
}

File: query.go
package altalune

type SortOrder string

const (
	SortOrderAsc  SortOrder = "asc"
	SortOrderDesc SortOrder = "desc"
)

type SortingParams struct {
	Field string
	Order SortOrder
}

type PaginationParams struct {
	Page     int32
	PageSize int32
}

type QueryParams struct {
	Pagination PaginationParams
	Keyword    string
	Filters    map[string][]string
	Sorting    *SortingParams
}

type QueryResult[T interface{}] struct {
	Data       []*T
	TotalRows  int32
	TotalPages int32
	Filters    map[string][]string
}
